### ISSUE TITLE
0.16.1.0 echo 'TODO run cabal new-install' TODO run cabal new-install

### ISSUE THREAD

**Issue by andrewufrank** [leksah/leksah#408]
i have problems when a module is compiled and tested but it seems not to be installed.
i see
echo 'TODO run cabal new-install'
TODO run cabal new-install

but then the next build fails with 'dependency tree exhaustively searched" 

i fear something substantial is missing for regular use?

**Comment by hamishmack** [comment]
Now we are using `cabal new-build` by default packages do not need to be installed to be reused.  Instead create a `cabal.project` file in a parent directory that is common to the packages that you want to work on together.  In this file list the directories of the the packages that you want to be built together.   Let's say you have two packages in your workspace and the cabal files are `something/package1dir/package1.cabal` and `something/package2dir/package2.cabal`.  Create a `something/cabal.project` file that looks like this:

```
packages:
  package1dir/
  package2dir/
```

When leksah runs `cabal new-build` in either of those two packages cabal will look for a `cabal.project` (starting in the directory of the package and then looking in its parents).  It will find `something/cabal.project` and build and install both packages into a package database in a `something/dist-newstyle` directory.

**Comment by andrewufrank** [comment]
i had problems with the stack build approach, because i have to include a stack.yaml in each subproject. i end up with situations where i can run stack build i a subproject and it is clean, but in leksah i get errors reported. i will try now with cabal new-build

**Comment by andrewufrank** [comment]
i removed all stack.yaml and created the cabal.project file as you indicated but i cannot get the cabal new-build to work in leksah: 

this is a trivial project 

```
[cabal new-build 'testPipes:lib:testPipes' 'testPipes:exe:simpler' 'testPipes:exe:test0'
cabal: Use of GHC's environment variable GHC_PACKAGE_PATH is incompatible with
Cabal. Use the flag --package-db to specify a package database (it can be used
multiple times).](url)
```

**Comment by andrewufrank** [comment]
back on using stack: there is a consistent problem that changes are not propagated up in the interfaces (i.e. a x not defined error, when x was introduced in an imported package). 
i guess the issue is mostly due to the stack.yaml files which i need in every module to avoid that cabal new-build is called. 
-- i used leksah 16.2.2 from github with commit  441e6ae  -- i assume you have a newer version.

**Comment by andrewufrank** [comment]
solved with 17.0.0
